crates/viewer/re_arrow_ui/src/datatype_ui.rs

```diff
@@ -0,0 +1,132 @@
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::FixedSizeList(field, size) => DataTypeUi {
+            type_name: format!("fixed-size list ({size})"),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::LargeList(field) => DataTypeUi {
+            type_name: "large list".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::LargeListView(field) => DataTypeUi {
+            type_name: "large list view".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, field);
+            })),
+        },
+        DataType::Union(fields, mode) => {
+            let label = match mode {
+                arrow::datatypes::UnionMode::Sparse => "sparse union",
+                arrow::datatypes::UnionMode::Dense => "dense union",
+            };
+            DataTypeUi {
+                type_name: label.to_owned(),
+                content: Some(Box::new(move |ui| {
+                    for (_, field) in fields.iter() {
+                        data_type_field_ui(ui, field);
+                    }
+                })),
+            }
+        }
+        DataType::Dictionary(_k, _v) => DataTypeUi {
+            type_name: "dictionary".to_owned(),
+            content: None,
+        },
+        DataType::Map(a, _) => DataTypeUi {
+            type_name: "map".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, a);
+            })),
+        },
+        DataType::RunEndEncoded(a, b) => DataTypeUi {
+            type_name: "run-end encoded".to_owned(),
+            content: Some(Box::new(move |ui| {
+                data_type_field_ui(ui, a);
+                data_type_field_ui(ui, b);
+            })),
+        },
+        non_nested => {
+            let label = simple_data_type_string(non_nested)
+                .map(|s| s.to_owned())
+                .unwrap_or_else(|| non_nested.to_string());
+            DataTypeUi {
+                type_name: label,
+                content: None,
+            }
+        }
+    }
+}
+
+fn data_type_field_ui(ui: &mut egui::Ui, field: &arrow::datatypes::Field) {
+    ui.spacing_mut().item_spacing.y = 0.0;
+    let item = ui.list_item();
+
+    let data_type_ui = data_type_ui(field.data_type());
+
+    let property = PropertyContent::new(field.name())
+        .value_text(data_type_ui.type_name)
+        .show_only_when_collapsed(false);
+
+    if let Some(content) = data_type_ui.content {
+        item.show_hierarchical_with_children(ui, Id::new(field.name()), true, property, content);
+    } else {
+        item.show_hierarchical(ui, property);
+    }
+}
+
+// TODO(emilk): there is some overlap here with `re_format_arrow`.
```

Comment:
It also has a lot of overlap with `crates/build/re_types_builder/src/codegen/docs/datatype_docs.rs`.

AFAIK there is no standard way of writing out arrow datatypes (e.g. `List<u8>` or `List[uint8]` etc), but it would be good if at least _we_ had a standard. Maybe you can create a quick issue about that and create a TODO and link to it from codegen, re_arrow_ui, and re_format_arrow? 🙏